File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/WarehouseStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class WarehouseStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'warehouseId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var WarehouseStateEventId
     */
    private $value;

    /**
     * @param WarehouseStateEventId $value
     */
    public function __construct(WarehouseStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new WarehouseStateEventId();
        }
    }

    /**
     * @return WarehouseStateEventId
     */
    public function toWarehouseStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getWarehouseId()
    {
        return $this->value->getWarehouseId();
    }

    /**
     * @param string $warehouseId
     */
    public function setWarehouseId($warehouseId)
    {
        $this->value->setWarehouseId($warehouseId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getWarehouseId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return WarehouseStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setWarehouseId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/WarehouseStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class WarehouseStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "warehouseId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private WarehouseStateEventId value;

    public WarehouseStateEventIdFlattenedDto()
    {
        this(new WarehouseStateEventId());
    }

    public WarehouseStateEventIdFlattenedDto(WarehouseStateEventId value)
    {
        this.value = value;
    }

    public WarehouseStateEventId toWarehouseStateEventId()
    {
        return this.value;
    }


    public String getWarehouseId()
    {
        return this.value.getWarehouseId();
    }

    public void setWarehouseId(String warehouseId)
    {
        this.value.setWarehouseId(warehouseId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        WarehouseStateEventIdFlattenedDto other = (WarehouseStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

